aitinero/src/app/providers/FormProvider.tsx

```diff
@@ -5,17 +5,32 @@ import { SubmitHandler, FormProvider as ReactHookFormProvider } from "react-hook
 import { FormValues } from "../types/form";
 import { useRouter } from "next/navigation";
 
+// BE imports
+import { useMutation } from "convex/react";
+import { api } from "../../../convex/_generated/api";
+
 export default function FormProvider({ children }: FormProviderProps) {
     const route = useRouter();
     const methods = useAppForm({
         location: '',
         startDate: undefined,
         endDate: undefined
     });
+    const createTrip = useMutation(api.trips.create);
 
-    const onSubmit: SubmitHandler<FormValues> = (data) => {
+    const onSubmit: SubmitHandler<FormValues> = async (data) => {
         const isValid = true;
+
         if (isValid) {
+            const id = await createTrip({
+                body: {
+                    location: data.location,
+                    start_date: new Date(data.startDate).toISOString(),
+                    end_date: new Date(data.endDate).toISOString(),
```

Comment:
Might be better if we make it into Date before we reach here since the typing for start_date and end_date is Date, but I think you understand how to do that better!